libhoney.go

```diff
@@ -159,10 +161,64 @@ type Builder struct {
 }
 
 type fieldHolder struct {
-	data map[string]interface{}
+	data marshallableMap
 	lock sync.Mutex
 }
 
+// Wrapper type for custom JSON serialization: individual values that can't be
+// marshalled (or are null pointers) will be skipped, instead of causing
+// marshalling to raise an error.
+type marshallableMap map[string]interface{}
+
+func (m marshallableMap) MarshalJSON() ([]byte, error) {
+	keys := make([]string, len(m))
+	i := 0
+	for k, _ := range m {
```

Comment:
style tip: if you only supply one variable on the left hand side of a `:= range`, it'll be the key. (aka, you don't need the `, _` if you don't want it)